main/main.c

```diff
@@ -116,6 +122,27 @@ void guiTask(void *pvParameter) {
     ESP_ERROR_CHECK(esp_timer_create(&periodic_timer_args, &periodic_timer));
     ESP_ERROR_CHECK(esp_timer_start_periodic(periodic_timer, LV_TICK_PERIOD_MS * 1000));
 
+    /* Create the demo application */
+    create_demo_application();
+    
+    while (1) {
+        vTaskDelay(1);
+
+        /* Try to take the semaphore, call lvgl related function on success */
+        if (pdTRUE == xSemaphoreTake(xGuiSemaphore, (TickType_t) LV_TICK_PERIOD_MS)) {
```

Comment:
I'm not sure if these lines are doing what you intend. It looks like you want to ensure that `lv_task_handler` gets called at least every `LV_TICK_PERIOD_MS` milliseconds. The current code looks like it will call `lv_task_handler` every 1ms as long as the semaphore is available.

I think this would make more sense:

```suggestion
        vTaskDelay(LV_TICK_PERIOD_MS);

        /* Try to take the semaphore, call lvgl related function on success */
        if (pdTRUE == xSemaphoreTake(xGuiSemaphore, portMAX_DELAY)) {
```

Since there's nothing else the task does if the semaphore is unavailable, it seems to me that it may as well just suspend until the semaphore becomes available.

Reply:
You're right @embeddedt, fixed it with 0b8b819.